app/scripts/utils/table-config/collections.js

```diff
@@ -43,14 +43,11 @@ export const tableSortProps = [
 ];
 
 const confirmRecover = (d) => `Recover ${d} ${strings.collection}(s)?`;
-export const recoverAction = function (collections) {
+export const recoverAction = function (collections, config) {
   return [{
     text: 'Recover',
-    action: (collectionId) => {
-      const { name, version } = collectionNameVersion(collectionId);
-      return recoverCollection(name, version);
-    },
-    state: collections.recovered,
+    action: config.recover.action,
+    state: collections.executed, // this will probably need to be changed
```

Comment:
What does this do?

Reply:
This lets the button know that it's state (SUCCESS/FAILED/IN_FLIGHT) is based on `collections.executed`. `collections.executed` is the state variable that is updated by `COLLECTION_APPLYWORKFLOW` actions.

It's worth noting that `granule` works the same way - we set `state: granules.executed` so the button pop-up/status thing takes a queue from the right state variable.

Reply:
Got it. This seems to be another result of how batch async commands work: https://github.com/nasa/cumulus-dashboard/blob/master/app/scripts/components/form/batch-async-command.js#L39